### ISSUE TITLE
Weird compilation bug

### ISSUE THREAD

**Issue by matthias-vogt** [sass/libsass#2168]
input.scss
```scss
@function test($a, $b) {
  @return "#{$a}" + "#{$b}" + "bar" + "bol";
}

foo {
  content: test('bim', 'baz');
}
```

Errors in libsass 3.3.6 (on sassmeister) and node-sass 3.10.7 (Edit: actually 3.8.0) on macOS 10.12 Beta (16A313a) with `Illegal instruction: 4` (error code 132)

**Comment by xzyfer** [comment]
I am unable to reproduce this with LibSass 3.3.6 on El Capitan. Unfortunately I don't have access to macOS. It's possible Apple has changed something that's affect LibSass, or there is an issue with the beta. Either way it's not something we can justify committing resources to until macOS is stable.

As far as I know there is no node-sass@v3.10.7. I know this because I released v3.9.0 5 minutes ago.

**Comment by xzyfer** [comment]
As an aside I have noticed related bug.

```sass
@function test($a, $b) {
  @return "#{$a}" + "#{$b}" + "" + "";
}

foo {
  content: test('bim', 'baz');
}
```
Ruby Sass
```
foo {
  content: "bimbaz"; }
```
LibSass
```
foo {
  content: "bimbazbaz"; }
```

**Comment by matthias-vogt** [comment]
Congrats on the release :) Sorry, i confused the version with my npm version. It's actually 3.8.0.
You can reproduce it by using [sassmeister](http://www.sassmeister.com/), copy pasting the code and changing the compiler to libsass in the options. (it won't let me save the code as a gist for some reason)

**Comment by xzyfer** [comment]
Apologies you're correct. I am now able to reproduce this with 3.3.6. This has been fixed in master and will be in 3.3.7.

**Comment by matthias-vogt** [comment]
Well that was quick, thanks a lot!
